mathesar/models.py

```diff
@@ -314,8 +316,12 @@ def delete_record(self, id_value):
         return delete_record(self._sa_table, self.schema._sa_engine, id_value)
 
     def add_constraint(self, constraint_type, columns, name=None):
+        if columns == []:
+            message = 'Columns field cannot be empty.'
+            raise base_api_exceptions.NotFoundAPIException(SyntaxError, message=message, status_code=status.HTTP_400_BAD_REQUEST)
         if constraint_type != constraint_utils.ConstraintType.UNIQUE.value:
-            raise ValueError('Only creating unique constraints is currently supported.')
+            message = 'Only creating unique constraints is currently supported.'
+            raise base_api_exceptions.ValueAPIException(ValueError, message=message, status_code=status.HTTP_400_BAD_REQUEST)
```

Comment:
@silentninja This is the change I originally made before #1375 got merged.

Reply:
@silentninja Passing an invalid type in the request still results in an Internal Server Error.